src/main/java/net/pms/util/ImagesUtil.java

```diff
@@ -145,4 +155,98 @@ public static void parseImageByImaging(File file, DLNAMediaInfo media) throws Im
 			media.setCodecV(FormatConfiguration.TIFF);
 		}
 	}
+
+	/**
+	 * Add the format(container) name of the media to the generic icon image.
+	 *
+	 * @param image BufferdImage to be the label added
+	 * @param label the media container name to be added as an label
+	 * @param renderer the renderer configuration
+	 * 
+	 * @return the generic icon with the container label added and scaled in accordance with renderer setting
+	 */
+	public static InputStream addFormatLabelToImage(BufferedImage image, String label, RendererConfiguration renderer) throws IOException {
+		// copy image to not affect the original one
+		ColorModel cm = image.getColorModel();
+		BufferedImage img = new BufferedImage(cm, image.copyData(null), cm.isAlphaPremultiplied(), null);
+
+		Graphics2D g = img.createGraphics();
+		g.drawImage(img, 0, 0, null);
+		g.setColor(Color.WHITE);
+		g.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 36));
+		g.drawString(label.toUpperCase(), 51, 62);
+		// XXX Is it necessary to use jpeg thumbnail-video with size 120x120 like it is used in the DLNAResource.getGenericThumbnailInputStream() method?
+		if (renderer != null) {
+			g.scale(renderer.getThumbnailWidth() / img.getWidth(), renderer.getThumbnailHeight() / img.getHeight());
```

Comment:
This is a bug, the division result will be an integer (any decimals will be trucated). That will be very unprecise. One of the division arguments should be cast to ```double``` before division to retain fractional value.

Here is FindBugs' explanation:
> This code casts the result of an integral division (e.g., int or long division) operation to double or float. Doing division on integers truncates the result to the integer value closest to zero. The fact that the result was cast to double suggests that this precision should have been retained. What was probably meant was to cast one or both of the operands to double before performing the division. Here is an example: 
```java
int x = 2;
int y = 5;
// Wrong: yields result 0.0
double value1 =  x / y;
>
>// Right: yields result 0.4
>double value2 =  x / (double) y;
```